OasisDEX/oasis-borrow
# Auto generated multipliers range

# [Auto generated multipliers range](https://app.shortcut.com/oazo-apps/story/5469/auto-generated-multipliers-range)

Updated list of possible multipliers available to choose from while setting up a Constant Multiple. Most of the ilks have static list of valid multipliers, as a fallback list generates automatically.

![image](https://user-images.githubusercontent.com/16230404/182621702-fa0fd258-df7b-40f2-ac53-be642e749649.png)
![image](https://user-images.githubusercontent.com/16230404/182625616-633bdef7-a811-481b-a480-71d4253ae1d1.png)
  
## Changes 👷‍♀️

- Added static lists for specifics ilks,
- created a method for auto generating as a fallback,
- refined, simplified and straighten some parts of the code related to CM UI changes and calculations,
- updated visuals to match figma designs,
  
## How to test 🧪

- Try to setup CM for vaults belonging to different ilks,
- please note that I DIDN'T test all the ilks if multiplier values are correct, I believe that should be a different story, I used values provided in here: https://www.notion.so/oazo/Ilk-multiples-1c616773bb9540b29e6124ff0955567c.
